pkg/internal/util/patch/toml_test.go

```diff
@@ -0,0 +1,198 @@
+			Name: "invalid 6902 patch JSON",
+			ToPatch: `disabled_plugins = ["restart"]
+[plugins.linux]
+  shim_debug = true
+[plugins.cri.containerd.runtimes.runsc]
+  runtime_type = "io.containerd.runsc.v1"`,
+			PatchesJSON6902: []string{"🏰"},
+			ExpectError:     true,
+		},
+		{
+			Name: "trivial patch",
+			ToPatch: `disabled_plugins = ["restart"]
+[plugins.linux]
+  shim_debug = true
+[plugins.cri.containerd.runtimes.runsc]
+  runtime_type = "io.containerd.runsc.v1"`,
+			Patches:     []string{`disabled_plugins=[]`},
+			ExpectError: false,
+			ExpectOutput: `[plugins]
+  [plugins.cri]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.runtimes]
+        [plugins.cri.containerd.runtimes.runsc]
+          runtime_type = "io.containerd.runsc.v1"
+  [plugins.linux]
+    shim_debug = true
+`,
+		},
+		{
+			Name: "trivial 6902 patch",
+			ToPatch: `disabled_plugins = ["restart"]
+[plugins.linux]
+  shim_debug = true
+[plugins.cri.containerd.runtimes.runsc]
+  runtime_type = "io.containerd.runsc.v1"`,
+			PatchesJSON6902: []string{`[{"op": "remove", "path": "/disabled_plugins"}]`},
+			ExpectError:     false,
+			ExpectOutput: `[plugins]
+  [plugins.cri]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.runtimes]
+        [plugins.cri.containerd.runtimes.runsc]
+          runtime_type = "io.containerd.runsc.v1"
+  [plugins.linux]
+    shim_debug = true
+`,
+		},
+		{
+			Name: "trivial patch and trivial 6902 patch",
+			ToPatch: `disabled_plugins = ["restart"]
+[plugins.linux]
+  shim_debug = true
+[plugins.cri.containerd.runtimes.runsc]
+  runtime_type = "io.containerd.runsc.v1"`,
+			Patches:         []string{`disabled_plugins=["foo"]`},
+			PatchesJSON6902: []string{`[{"op": "remove", "path": "/disabled_plugins"}]`},
+			ExpectError:     false,
+			ExpectOutput: `[plugins]
+  [plugins.cri]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.runtimes]
+        [plugins.cri.containerd.runtimes.runsc]
+          runtime_type = "io.containerd.runsc.v1"
+  [plugins.linux]
+    shim_debug = true
+`,
+		},
+		{
+			Name: "invalid path 6902 patch",
+			ToPatch: `disabled_plugins = ["restart"]
+[plugins.linux]
+  shim_debug = true
+[plugins.cri.containerd.runtimes.runsc]
+  runtime_type = "io.containerd.runsc.v1"`,
+			PatchesJSON6902: []string{`[{"op": "remove", "path": "/fooooooo"}]`},
+			ExpectError:     true,
+			ExpectOutput: `[plugins]
+  [plugins.cri]
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.runtimes]
+        [plugins.cri.containerd.runtimes.runsc]
+          runtime_type = "io.containerd.runsc.v1"
+  [plugins.linux]
+    shim_debug = true
+`,
+		},
+		{
+			Name: "patch registry",
+			ToPatch: `disabled_plugins = ["restart"]
+[plugins.linux]
+  shim_debug = true
+[plugins.cri.containerd.runtimes.runsc]
+  runtime_type = "io.containerd.runsc.v1"`,
+			Patches: []string{`[plugins.cri.registry.mirrors]
+  [plugins.cri.registry.mirrors."registry:5000"]
+    endpoint = ["http://registry:5000"]`},
```

Comment:
sample data from a real end-user work-around in #110 (kubevirt is doing this with `sed` at runtime today)

Reply:
(which will be increasingly tricky for them to do correctly as we configure more things in the base config, applying a patch avoids that problem)